### ISSUE TITLE
Allow using sysimage-powered REPL and building a new sysimage simultaneously

### ISSUE THREAD

**Issue by ericphanson** [julia-vscode/julia-vscode#1161]
This permits a workflow of after you add some packages, you can use them as normal in your current sysimage-powered-REPL (assuming the versions of dependencies etc are compatible with those already loaded-- as usual regardless of sysimages) but also can kick off a build so next time you restart the session they’ll be baked into the sysimage already.

My thought for the implementation would be to just name newly-built sysimages with a `new` suffix (or similar), and then have the `start REPL` invocation check for a new `sysimage`, and if it exists, replace the the current one with it. (So the REPL always uses a current one if it exists, but if there's a new one then just replace the current one first before starting up).

As @davidanthoff pointed out in Slack such a system would also allow a failure in building the sysimage to not clobber the working one (I guess there should be a cleanup step that if there is a failure then don't create any `new`-suffixed sysimage so it doesn't clobber the working one when the REPL starts up, and also please be vocal so I know why when I restart the REPL I don't have the new packages / versions loaded!).

Other considerations:
* If one was just adding new packages which use the same versions of any common dependencies, one could consider using incrementally adding to the current sysimage. I would think one wants to be very careful about doing something like that, because often when you add a new package the resolver might change the versions of some dependencies to allow you to add it (especially in 1.4+), and the old versions will already be baked into the sysimage. However with lots of Manifest tracking maybe it could be possible to see when it's safe to do so and speed up builds that way.
* I don't think this should be automatic because it's a big job. However even with it being manual it can make a very smooth workflow where you only have to pay the cost of `using` times once (i.e. when you load the package into your old sysimage to keep working while the new one is built, but not afterwards if you restart the session). And with https://github.com/julia-vscode/julia-vscode/issues/1160 you also can just pay first-run costs once too by copying over any extensive-to-compile commands to a execution script.

**Comment by davidanthoff** [comment]
The only downside I can think of right now is a bit of a weird scenario like this: you have two VS Code windows open, both with an open REPL that uses the current sysimage. You compile a new sysimage in one window, when it is done you close the REPL and reopen it, expecting to get the new sysimage. But you won't because the other window still has the old sysimage open and so the new one can't be copied over. So that is a bit weird, I think... Not sure how we could get around that.

In any case, the code places that one would probably have to change are [here](https://github.com/julia-vscode/julia-vscode/blob/b7a8b6b3fa838a64715f2315d634f2d9d09f79c9/scripts/tasks/task_compileenv.jl#L5), [here](https://github.com/julia-vscode/julia-vscode/blob/b7a8b6b3fa838a64715f2315d634f2d9d09f79c9/src/repl.ts#L193) and [here](https://github.com/julia-vscode/julia-vscode/blob/b7a8b6b3fa838a64715f2315d634f2d9d09f79c9/src/jlpkgenv.ts#L21).

**Comment by ericphanson** [comment]
Ah, true. Maybe just some kind of lock file to ensure we don't overwrite an active sysimage, and then throw a warning when there is a new sysimage but we can't overwrite so we are using the old one? Or if we can't overwrite the old one then throw a warning to say so and then just start up with the new one?

**Comment by davidanthoff** [comment]
I don't think we need a lock file, right? I think we would just need an error message if the rename step from `-new` file to the real sysimage file failed, saying "please first close all REPLs" or something like that?

**Comment by ericphanson** [comment]
Does it actually fail though? I tried creating a sysimage while using a sysimage, and didn't see any errors or such.

**Comment by davidanthoff** [comment]
There is probably a difference between Windows (where I'm pretty sure a sysimg file is locked while in use) and other platforms? Not entirely sure how that works on non Windows...
